src/Microsoft.AspNetCore.Sockets/Internal/ChannelConnection.cs

```diff
@@ -11,20 +11,23 @@ namespace Microsoft.AspNetCore.Sockets.Internal
 {
     public class ChannelConnection<T> : IChannelConnection<T>
     {
-        public IReadableChannel<T> Input { get; }
-        public IWritableChannel<T> Output { get; }
+        public IChannel<T> Input { get; }
```

Comment:
Why do we need to expose both sides of both channels here? I've been trying to avoid that as much as possible...